src/routers/asset.ts

```diff
@@ -133,6 +138,27 @@ export function handle(context: IndexerContext, router: Router) {
             const confirmThreshold =
                 req.query.confirmThreshold &&
                 parseInt(req.query.confirmThreshold, 10);
+            let lastEvaluatedKey;
+            {
+                try {
+                    lastEvaluatedKey =
+                        req.query.lastEvaluatedKey &&
+                        JSON.parse(req.query.lastEvaluatedKey);
+                    if (lastEvaluatedKey) {
+                        if (lastEvaluatedKey.length !== 3) {
+                            throw new Error("Invalid lastEvalulatedKey");
+                        }
+                        for (let index = 0; index < 3; index++) {
+                            if (!Number.isInteger(lastEvaluatedKey[index])) {
+                                throw new Error("Invalid lastEvalulatedKey");
+                            }
+                        }
+                    }
+                } catch (e) {
+                    next(e);
+                }
+            }
+
```

Comment:
I want to use `Joi` to verify the `lastEvaluatedKey`. However, `Joi` does not support `JSON` string. @joojis Do you have any idea?

Reply:
I tried using express middleware. What do you think about using express middleware for this purpose?

https://github.com/CodeChain-io/codechain-indexer/pull/340/commits/9f15eb261616179106c61ecd26270ab6b01e3a29

Reply:
Doesn't it support writing custom functions like `() -> bool`?

Reply:
Oh, it seems it doesn't.

Reply:
Oops, sorry. There is an `any.custom()`. https://hapi.dev/family/joi/?v=16.1.7#anycustommethod-description

Reply:
Aha. I'll try it. Thanks!

Reply:
`any.custom()` needs version 16 of `joi`. I tried to update the `joi` library but failed.